jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -24,9 +24,58 @@
 from jdaviz.core.tools import ICON_DIR
 from jdaviz.core.custom_traitlets import IntHandleEmpty
 
+from scipy.interpolate import make_interp_spline
+
 __all__ = ['PlotOptions']
 
 
+class SplineStretch:
+    """
+    A class to represent spline stretches.
+
+    Attributes:
+        k (int): Degree of the smoothing spline.
+        bc_type (str or None): Boundary condition type.
+        t (array-like or None): Array of knot positions.
+        x (array-like): The x-coordinates of the data points.
+        y (array-like): The y-coordinates of the data points.
+        spline (object): Interpolating spline.
+    """
+
+    def __init__(self):
+        """
+        Initializes the SplineStretch class with default knot values.
+        """
+        self.k = 3
+        self.bc_type = None
+        self.t = None
+        self.update_knots(
+            x=np.array([0, 0.1, 0.2, 0.7, 1]),
+            y=np.array([0, 0.05, 0.3, 0.9, 1])
+        )
+
+    def __call__(self, values, out=None, clip=False):
+        # TODO: Review the 'out' and 'clip' parameters once glue upstream is updated.
+        # Depending on their implementation, this method may need adjustment.
+        return self.spline(values)
+
+    def update_knots(self, x, y):
+        if len(x) != len(y):
+            raise ValueError("x and y must be the same length.")
+        self.x = x
+        self.y = y
+        self._update_spline()

```

Comment:
Is it really necessary to put a one-line calculation into a different method? Do you plan to do lazy evaluation or something?

Reply:
I used the _update_spline() method to encapsulate the spline update process, thinking it might become more complex in the future or might be reused in other parts of the code. However, I see your point. Since it's currently a single line, I can integrate it directly for clarity and simplicity. Regarding lazy evaluation, I hadn't planned on it, but I'm open to suggestions if you think it'd be beneficial in this context.

Reply:
Lazy evaluation might not be worth it unless you start seeing performance problems. The thing with being lazy is that it could be *too lazy*, leading to inaccurate results, kinda like the cache invalidation problem. So probably unnecessary for now.